src/renderer/components/NodeGridElement.tsx

```diff
@@ -48,8 +50,8 @@ const NodeGridElement: React.FC<NodeGridElementProps> = ({ node, graphName, temp
     const isHighContrast = useSelector(getHighContrastState);
     const showLinkSaturation = useSelector(getShowLinkSaturation);
 
-    // TODO: pre-calculate link saturation and memoize it
-    const linksData = useSelector(getAllLinksForGraph(graphName));
+    const linksData = useSelector(getNodeLinksData(temporalEpoch, node.uid));
+    const nodeSaturation = useSelector(getOffchipNodeSaturation(temporalEpoch, node.uid));
```

Comment:
maybe `offchipLinkSaturation`  and `offchipLinkSaturationForNode` as a selector?